pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDTask.java

```diff
@@ -130,7 +131,8 @@ private void report(CPD cpd) throws ReportException {
         if (!cpd.getMatches().hasNext()) {
             log("No duplicates over " + minimumTokenCount + " tokens found", Project.MSG_INFO);
         }
-        CPDRenderer renderer = createRenderer();
+        CPDReportRenderer renderer = createRenderer();
+        CPDReport report = new CPDReport(cpd.getMatches(), cpd.getNumberOfTokensPerFile());
```

Comment:
I think it would be nicer to make a method on `CPD` that encapsulates that:
```suggestion
        CPDReport report = cpd.toReport();
```